src/nifgen/system_tests/test_system_nifgen.py

```diff
@@ -244,18 +244,65 @@ def test_write_binary_waveform(session):
     session.write_binary16_waveform(session.allocate_waveform(10), [0, 0, 0, 1, 1, 1, 2, 2])
 
 
-'''
-(TODO) Jaleel , check it after issue #538 fixed
 def test_set_waveform_next_write_position(session):
-    session.set_waveform_next_write_position(session.allocate_waveform(10), nifgen.RelativeTo.START, 5)  # Enable after RelativeTo enum added to enums_addon.py
+    session.set_waveform_next_write_position(session.allocate_waveform(10), nifgen.RelativeTo.START, 5)
 
 
 def test_export_signal(session):
     expected_trigger_terminal = "PXI_Trig0"
-    session.export_signal(nifgen.ExportSignal.START_TRIGGER, "", expected_trigger_terminal)  # Enable after issue #538 fixed
+    session.export_signal(nifgen.Signal.START_TRIGGER, "", expected_trigger_terminal)
     assert expected_trigger_terminal == session.exported_start_trigger_output_terminal
 
 
 def test_write_waveform_from_filei64(session):
-    session.create_waveform_from_file_i16(os.path.join(os.getcwd(), 'SineI16BigEndian_1000.bin'), nifgen.ByteOrder.BIG_ENDIAN)  # Enable after issue #538 fixed
+    session.create_waveform_from_file_i16(os.path.join(os.getcwd(), 'src\\nifgen\\system_tests', 'SineI16BigEndian_1000.bin'), nifgen.ByteOrder.BIG)
+
+
+def test_named_waveform_operations(session):
+    wfm_name = 'Waveform'
+    wfm_size = 4096
+    write_offset = 0
+    wfm_data_f64 = [0.0, 1.0]
+    wfm_data_i16 = [1, 0]
+    session.allocate_named_waveform(wfm_name, wfm_size)
+    session.set_named_waveform_next_write_position(wfm_name, nifgen.RelativeTo.START, write_offset)
+    session.write_named_waveform_f64(wfm_name, wfm_data_f64)
+    session.write_named_waveform_i16(wfm_name, wfm_data_i16)
+    session.delete_named_waveform(wfm_name)
+
+
+def test_wait_until_done(session):
+    session.wait_until_done(20)
+
+
+def test_user_standard_waveform(session):
+    wfm_points = [1] * 8192
+    session.output_mode = nifgen.OutputMode.FUNC
+    session.configure_standard_waveform(nifgen.Waveform.USER, 1.0, 2000000, 1.0, 0.0)
+    session.define_user_standard_waveform(wfm_points)
+    session.clear_user_standard_waveform()
+
+
+# TODO(bhaswath): Doesn't work, issue #596
+'''
+def test_fir_filter_coefficients():
+    with nifgen.Session('', False, 'Simulate=1, DriverSetup=Model:5441;BoardType:PXI') as session:
+        coeff_array = [1, 0, -1]
+        session.configure_custom_fir_filter_coefficients(coeff_array)
+        session.commit()
+        array, size = session.get_fir_filter_coefficients()
+        assert size == len(coeff_array)
+'''
+
+
+def test_configure_triggers(session):
+    session.configure_digital_edge_start_trigger('PFI0', nifgen.StartTriggerDigitalEdgeEdge.FALLING)
+    session.configure_digital_level_script_trigger('ScriptTrigger0', 'PXI_Trig0', nifgen.TriggerWhen.HIGH)
+
+
+# TODO(bhaswath): Enable after Issue 597 is fixed
```

Comment:
You added all except one enum in this PR, maybe you can finish it and close #597 too?